src/main/scala/org/clulab/wm/eidos/groundings/EidosWordToVec.scala

```diff
@@ -75,6 +79,28 @@ class RealWordToVec(val w2v: CompactWord2Vec, topKNodeGroundings: Int) extends E
     }
   }
 
+  def calculateSimilaritiesWeighted(canonicalNameParts: Array[String], posTags:Seq[String], nounVerbWeightRatio:Float, conceptEmbeddings: Seq[ConceptEmbedding]): EidosWordToVec.Similarities = {
+    val sanitizedNameParts = canonicalNameParts.map(Word2Vec.sanitizeWord(_))
+    // It could be that the composite vectore below has all zeros even though some values are defined.
+    // That wouldn't be OOV, but a real 0 value.  So, conclude OOV only if none is found (all are not found).
+    val outOfVocabulary = sanitizedNameParts.forall(w2v.isOutOfVocabulary(_))
+    val termWeights = posTags.map(tag=>if (tag.startsWith("NN")) nounVerbWeightRatio else 1.0.toFloat)
+
+
+    if (outOfVocabulary)
+      Seq.empty
+    else {
+      val nodeEmbedding = w2v.makeCompositeVectorWeighted(sanitizedNameParts, termWeights)
```

Comment:
@zhengzhongliang I'm trying to run `EvalGroundings` to try out your changes, but I run into an error here with `makeCompositeVectorWeighted`. It doesn't seem to exist anywhere. I pulled your branch directly, too.

Reply:
I'll be trying it out soon as well, but note step 4 above.  It requires a change to processors that has not yet been published, so doing that locally is required.

Reply:
I pinged (pang?) Zhengzhong on Slack for help with step 4 and I got things running.